app/main.py

```diff
@@ -1,25 +1,89 @@
-from abc import ABC
+from __future__ import annotations
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Visitor, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: Visitor, owner: Visitor) -> Visitor:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: Visitor, value: int) -> None:
```

Comment:
Place your validation logic in the `__set__` method here, it should make use of `self.min_amount` and `self.max_amount` attributes.

Reply:
This comment is not resolved, replace `Visitor` with `Any` in the type annotation for the `IntegerRange` class.